plugins/c9.vfs.client/vfs_client.js

```diff
@@ -193,12 +193,12 @@ define(function(require, exports, module) {
                 extraPaths = path;
                 path = path[0];
                 extraPaths = "," + extraPaths.map(function(p) {
-                    return p[0] == path[0] && p != path ? escape(p) : "";
+                    return p[0] == path[0] && p != path ? encodeURI(p) : "";
```

Comment:
@nightwing I have reproduced the issue where filenames including a `,` get interpreted as several filenames. I'll think about how to resolve this while still preserving the correct behaviour with filenames containing `+`.